ibis/backends/conftest.py

```diff
@@ -259,9 +441,29 @@ def pytest_runtest_call(item):
 
 
 @pytest.fixture(params=_get_backends_to_test(), scope='session')
-def backend(request, data_directory):
-    """Return an instance of BackendTest."""
+def backend(
+    request, data_directory, script_directory, tmp_path_factory, worker_id
+):
+    """Return an instance of BackendTest, loaded with data."""
+    from filelock import FileLock
+
     cls = _get_backend_conf(request.param)
+
+    # Load testdata once per session
+    if worker_id == "master":  # single-threaded pytest
+        cls.load_data(data_directory, script_directory)
+        return cls(data_directory)
+
+    # multi-threaded pytest
+
+    # get the temp directory shared by all workers
+    root_tmp_dir = tmp_path_factory.getbasetemp().parent
+
+    fn = root_tmp_dir / "lockfile"
```

Comment:
```suggestion
    fn = root_tmp_dir / "lockfile.lock"
```

Reply:
I think what I have is correct!

There is `lockfile` and `lockfile.lock`. If `lockfile.lock` does not exist and `lockfile` does not, create `lockfile.lock`, load data, and then create `lockfile`.
If `lockfile.lock` exists, wait until it does not exist, and then continue.
If `lockfile.lock` does not exist and `lockfile` does, then continue.

Reply:
Ahh right, so `lockfile` itself indicates whether the backend has been initialized or not. 

Is this `lockfile` shared for all backends? Do we need separate lockfiles perhaps?

Reply:
> Is this lockfile shared for all backends? Do we need separate lockfiles perhaps?

This is a really good point! I think we do.